src/DynamicRupture/Output/Base.cpp

```diff
@@ -0,0 +1,323 @@
+
+    auto ltsMap = faceToLtsMap[outputData.receiverPoints[i].faultFaceIndex];
+    const auto layer = ltsMap.first;
+    const auto ltsId = ltsMap.second;
+
+    auto mu = reinterpret_cast<DrPaddedArrayT>(layer->var(dynRup->mu));
+    auto rt = reinterpret_cast<DrPaddedArrayT>(layer->var(dynRup->ruptureTime));
+    auto slip = reinterpret_cast<DrPaddedArrayT>(layer->var(dynRup->slip));
+    auto peakSR = reinterpret_cast<DrPaddedArrayT>(layer->var(dynRup->peakSlipRate));
+
+    const auto nearestGaussPoint = outputData.receiverPoints[i].nearestGpIndex;
+    const auto faceIndex = outputData.receiverPoints[i].faultFaceIndex;
+
+    const auto normal = outputData.faultDirections[faceIndex].faceNormal;
+    const auto tangent1 = outputData.faultDirections[faceIndex].tangent1;
+    const auto tangent2 = outputData.faultDirections[faceIndex].tangent2;
+    const auto strike = outputData.faultDirections[faceIndex].strike;
+    const auto dip = outputData.faultDirections[faceIndex].dip;
+
+    auto& functionAndState = std::get<VariableID::FunctionAndState>(outputData.vars);
+    if (functionAndState.isActive) {
+      functionAndState(ParamID::FUNCTION, level, i) = mu[ltsId][nearestGaussPoint];
+    }
+
+    auto& ruptureTime = std::get<VariableID::RuptureTime>(outputData.vars);
+    if (ruptureTime.isActive) {
+      ruptureTime(level, i) = rt[ltsId][nearestGaussPoint];
+    }
+
+    auto& absoluteSlip = std::get<VariableID::AbsoluteSlip>(outputData.vars);
+    if (absoluteSlip.isActive) {
+      absoluteSlip(level, i) = slip[ltsId][nearestGaussPoint];
+    }
+
+    auto& peakSlipsRate = std::get<VariableID::PeakSlipsRate>(outputData.vars);
+    if (peakSlipsRate.isActive) {
+      peakSlipsRate(level, i) = peakSR[ltsId][nearestGaussPoint];
+    }
+
+    auto& slipVectors = std::get<VariableID::Slip>(outputData.vars);
+    if (slipVectors.isActive) {
+      VrtxCoords crossProduct = {0.0, 0.0, 0.0};
+      MeshTools::cross(strike, tangent1, crossProduct);
+
+      double cos1 = MeshTools::dot(strike, tangent1);
+      double scalarProd = MeshTools::dot(crossProduct, normal);
+
+      // Note: cos1**2 can be greater than 1.0 because of rounding errors -> min
+      double sin1 = std::sqrt(1.0 - std::min(1.0, cos1 * cos1));
+      sin1 = (scalarProd > 0) ? sin1 : -sin1;
+
+      auto slip1 = reinterpret_cast<DrPaddedArrayT>(layer->var(dynRup->slipStrike));
+      auto slip2 = reinterpret_cast<DrPaddedArrayT>(layer->var(dynRup->slipDip));
+
+      slipVectors(DirectionID::STRIKE, level, i) =
+          cos1 * slip1[ltsId][nearestGaussPoint] - sin1 * slip2[ltsId][nearestGaussPoint];
+
+      slipVectors(DirectionID::DIP, level, i) =
+          sin1 * slip1[ltsId][nearestGaussPoint] + cos1 * slip2[ltsId][nearestGaussPoint];
+    }
+  }
+
+  if (type == OutputType::AtPickpoint) {
+    outputData.cachedTime[outputData.currentCacheLevel] = time;
+    outputData.currentCacheLevel += 1;
+  }
+}
+
+void Base::tiePointers(seissol::initializers::Layer& layerData,
```

Comment:
Yes, you're correct. The original idea of `tiePointers` was to assign pointers of the memory allocated in C++ to Fortran pointers. But Adrian didn't have time and he just did a copy. Why is it bad? Because when we update Point Receivers (which is usually a dozen) located along the fault and tend to have higher output frequency we copy all fault elements to Fortran.   Aliasing pointers between C++ and Fortran is a better way.